server/registry/actions_apis.go

```diff
@@ -176,22 +176,22 @@ func (s *RegistryServer) UpdateApi(ctx context.Context, req *rpc.UpdateApiReques
 	var response *rpc.Api
 	if err := s.runInTransaction(ctx, func(ctx context.Context, db *storage.Client) error {
 		db.LockApis(ctx)
-		api, err := db.GetApi(ctx, name)
-		if err == nil {
-			if err := api.Update(req.GetApi(), models.ExpandMask(req.GetApi(), req.GetUpdateMask())); err != nil {
-				return err
-			}
-			if err := db.SaveApi(ctx, api); err != nil {
-				return err
-			}
-			response, err = api.Message()
-			return err
-		} else if status.Code(err) == codes.NotFound && req.GetAllowMissing() {
-			response, err = s.createApi(ctx, db, name, req.GetApi())
+		api, err := models.NewApi(name, req.GetApi())
+		if err != nil {
 			return err
-		} else {
+		}
+		mask := models.ExpandMask(req.GetApi(), req.GetUpdateMask()).GetPaths()
+		if err := db.SaveApi(ctx, api, mask); err != nil {
+			if status.Code(err) == codes.NotFound && req.GetAllowMissing() {
+				response, err = s.createApi(ctx, db, name, req.GetApi())
+				if status.Convert(err).Code() == codes.AlreadyExists {
```

Comment:
What's the difference between `status.Convert(err).Code()` and `status.Code(err)`? This is my first time seeing the convert method.

Reply:
From the [docs](https://pkg.go.dev/google.golang.org/grpc/status#Convert): `Convert is a convenience function which removes the need to handle the boolean return value from FromError.`

Reply:
Hmm. Looking at `status.Code(err)` now, perhaps I didn't answer your real question of why not use that? I'll try it. Seems like there are plenty of options in this package. LOL

Reply:
Ok, changed.